### PR TITLE
Fix 500 error when stackoverflow post is unanswered

### PR DESCRIPTION
**Pull request by irisliu77**

### CODE REVIEW

File: src/main/java/com/google/step/snippet/external/StackOverflowClient.java [hunk 1]
```diff
@@ -173,7 +174,11 @@ private String getResponse(String url, String fieldParam) {
       return null;
     }
     JSONObject json = new JSONObject(responseBody.toString());
-    String res = json.getJSONArray(ITEM_PARAMETER).getJSONObject(0).get(fieldParam).toString();
+    JSONArray items = json.getJSONArray(ITEM_PARAMETER);
+    if (items == null || items.length() == 0) {
+      return null;
+    }
+    String res = items.getJSONObject(0).get(fieldParam).toString();
     if (response.getStatusLine().getStatusCode() != 200) {
```
Comment (kaylaleung) [inline src/main/java/com/google/step/snippet/external/StackOverflowClient.java:182]: Can follow a similar style in SearchServlet #18 :

```
if (json.has(ITEM_PARAMTER) && json.getAsJsonArray(ITEM_PARAMTER).size() > 0) {
    JsonObject item = obj.getAsJsonArray(ITEM_PARAMTER).get(0).getAsJsonObject();
    if (item.has(fieldParam)) {
        String res = item.get(fieldParam).getAsString();
    }
}
```

File: src/main/java/com/google/step/snippet/external/StackOverflowClient.java [hunk 2]
```diff
@@ -173,7 +174,11 @@ private String getResponse(String url, String fieldParam) {
       return null;
     }
     JSONObject json = new JSONObject(responseBody.toString());
-    String res = json.getJSONArray(ITEM_PARAMETER).getJSONObject(0).get(fieldParam).toString();
+    JSONArray items = json.getJSONArray(ITEM_PARAMETER);
+    if (items == null || items.length() == 0) {
+      return null;
+    }
+    String res = items.getJSONObject(0).get(fieldParam).toString();
```
Comment (AaronLopes) [inline src/main/java/com/google/step/snippet/external/StackOverflowClient.java:181]: nit: change to "result" or something more readable

**Comment by anbenson** [comment]
Can you change the title to be a little more descriptive? I have to look up the issue each time :P

File: src/main/java/com/google/step/snippet/external/StackOverflowClient.java (hunk 1 above)
Comment (irisliu77) [reply src/main/java/com/google/step/snippet/external/StackOverflowClient.java:182]: both ways look fine i think..?

File: src/main/java/com/google/step/snippet/external/StackOverflowClient.java (hunk 2 above)
Comment (kaylaleung) [reply src/main/java/com/google/step/snippet/external/StackOverflowClient.java:181]: Unresolved?

File: src/main/java/com/google/step/snippet/external/StackOverflowClient.java (hunk 1 above)
Comment (kaylaleung) [reply src/main/java/com/google/step/snippet/external/StackOverflowClient.java:182]: Uh, defer to Andrew and Joan?

But I still don't think you're safely accessing the fieldParam here?

File: src/main/java/com/google/step/snippet/external/StackOverflowClient.java (hunk 2 above)
Comment (irisliu77) [reply src/main/java/com/google/step/snippet/external/StackOverflowClient.java:181]: uh alright, i just feel res is somewhat common

**Comment by kaylaleung** [comment]
SO card error seems fixed, but actually, the original issue opened in #39 is still not completely addressed, because of lack of null check for links #65

File: src/main/java/com/google/step/snippet/external/StackOverflowClient.java [hunk 3]
```diff
@@ -173,10 +174,14 @@ private String getResponse(String url, String fieldParam) {
...
     JSONObject json = new JSONObject(responseBody.toString());
-    String res = json.getJSONArray(ITEM_PARAMETER).getJSONObject(0).get(fieldParam).toString();
-    if (response.getStatusLine().getStatusCode() != 200) {
-      return null;
+    if (json.has(ITEM_PARAMETER) && json.getJSONArray(ITEM_PARAMETER).length() > 0) {
+      JSONArray items = json.getJSONArray(ITEM_PARAMETER);
+      JSONObject obj = items.getJSONObject(0);
+      if (obj.has(fieldParam)) {
+        String result = obj.get(fieldParam).toString();
+        return result;
```
Comment (kaylaleung) [inline src/main/java/com/google/step/snippet/external/StackOverflowClient.java:182]: can just directly `return obj.get(fieldParam).toString();` here

**Pull request merged**